File: src/main/java/one/oktw/i18n/api/provider/TranslationStringProvider.java
package one.oktw.i18n.api.provider;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Locale;

public interface TranslationStringProvider {
    @Nullable
    String get(@NotNull Locale var1, @NotNull String var2);
}

File: src/main/kotlin/one/oktw/i18n/api/provider/TranslationStringProvider.kt
package one.oktw.i18n.api.provider

import java.util.*

interface TranslationStringProvider {
    companion object {
        class ScopedTranslationStringProvider(private val scope: String, private val provider: TranslationStringProvider): TranslationStringProvider {
            override fun get(locale: Locale, key: String): String? {
                return if (key.startsWith("$scope:")) {
                    provider.get(locale, key.drop(scope.length + 1))
                } else {
                    null
                }
            }
        }
    }

    // get the translation key from key
    fun get(locale: Locale, key: String): String?
}